components/video-recorder/src/HeadlessVideoRecorder.tsx

```diff
@@ -0,0 +1,636 @@
+	const playbackRef = useRef<HTMLVideoElement>(null);
+	const mediaRecorderRef = useRef<MediaRecorder | null>(null);
+
+	const recordingTime = timeInMs;
+
+	const timer = useTimer();
+
+	const isSupportedMimeType = (mimeTypes: VideoMimeTypes[]) => {
+		const result: VideoMimeTypes[] = [];
+		mimeTypes.forEach((mimeType) => {
+			if (
+				typeof MediaRecorder !== "undefined" &&
+				MediaRecorder.isTypeSupported(mimeType)
+			) {
+				result.push(mimeType);
+			}
+		});
+		return result.length === 0 ? undefined : result[0];
+	};
+
+	const supportedMimeType = isSupportedMimeType(["video/mp4", "video/webm"]);
+	const mimeType = supportedMimeType;
+	const blobType = supportedMimeType;
+	const fileExtension = supportedMimeType?.split("/")[1];
+
+	const showStreamPreview = useCallback(async () => {
+		let mergeVideoConstraints: MediaStreamConstraints["video"] = {
+			width: { ideal: 640, max: 640 },
+			height: { ideal: 480, max: 480 },
+			facingMode: "user",
+			frameRate: { ideal: 15, max: 30 },
+		};
+
+		if (
+			typeof mediaConstraints?.video === "object" &&
+			mediaConstraints.video !== null
+		) {
+			mergeVideoConstraints = {
+				...mergeVideoConstraints,
+				...mediaConstraints.video,
+			};
+		}
+		const mergeAudioConstraints = mediaConstraints?.audio ?? true;
+
+		const stream = await navigator.mediaDevices.getUserMedia({
+			video: mergeVideoConstraints,
+			audio: mergeAudioConstraints,
+		});
+
+		if (videoRef.current) {
+			videoRef.current.srcObject = stream;
+			videoRef.current.onloadedmetadata = () => {
+				setLoading(false);
+			};
+		}
+	}, [videoRef]);
+
+	const showPreview = useCallback(async () => {
+		try {
+			setLoading(true);
+
+			showStreamPreview();
+			setRecording("preview");
+		} catch (error) {
+			console.log("Preview Error__", error);
+		}
+	}, [showStreamPreview]);
+
+	useEffect(() => {
+		autoPreview && showPreview();
+	}, [showPreview]);
+
+	useEffect(() => {
+		handleStateChange && handleStateChange(recording);
+	}, [recording]);
+
+	const handleFileDownload = useCallback(() => {
+		if (file) {
+			const url = URL.createObjectURL(file);
+			const a = document.createElement("a");
+			a.href = url;
+			a.download = file.name || "recorded-video";
+			document.body.appendChild(a);
+			a.click();
+			document.body.removeChild(a);
+			URL.revokeObjectURL(url);
+		} else {
+			console.log("File not Found!");
+		}
+	}, [file]);
+
+	const handleDownload = useCallback(() => {
+		(customhandleDownload || handleFileDownload)(file);
+	}, [file]);
+
+	useImperativeHandle(
+		ref,
+		() => {
+			return {
+				recording,
+				handleDownload,
+				showPreview,
+			};
+		},
+		[recording, handleDownload, showPreview],
+	);
+
+	const wait = (delayInMS: number) => {
+		return new Promise((resolve) => setTimeout(resolve, delayInMS));
+	};
+
+	const record = async (stream: MediaStream, lengthInMs: number) => {
+		const recorder = new MediaRecorder(stream, {
+			mimeType,
+			videoBitsPerSecond: 2500000,
```

Comment:
Should this hard coded value be editable? What does this do?

Reply:
videoBitsPerSecond is a property of the MediaRecorder interface that returns the video encoding bit rate in use,it determines the quality and size of the recorded video.
we can make it editable if required